crates/common/src/types/token_vault_flattened.rs

```diff
@@ -1,33 +1,47 @@
 use crate::csv::TryIntoCsv;
-use rain_orderbook_subgraph_client::types::vaults_list;
+use alloy_primitives::{utils::format_units, U256};
+use rain_orderbook_subgraph_client::types::vaults_list::*;
 use serde::{Deserialize, Serialize};
 
+use super::FlattenError;
+
 #[derive(Serialize, Deserialize, Clone)]
 pub struct TokenVaultFlattened {
     pub id: String,
-    pub owner: vaults_list::Bytes,
-    pub vault_id: vaults_list::BigInt,
-    pub token_name: String,
-    pub token_symbol: String,
-    pub token_decimals: i32,
+    pub owner: Bytes,
+    pub vault_id: BigInt,
+    pub token_name: Option<String>,
+    pub token_symbol: Option<String>,
+    pub token_decimals: Option<BigInt>,
```

Comment:
https://eips.ethereum.org/EIPS/eip-20#decimals decimals is actually a `u8`, not a `i32` nor a `BigInt`